pkgs/by-name/st/story-architect/package.nix

```diff
@@ -0,0 +1,51 @@
+{
+  stdenvNoCC,
+  lib,
+  fetchurl,
+  appimageTools,
+  makeWrapper,
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "story-architect";
+  version = "0.7.5";
+
+  src = fetchurl {
+    url = "https://github.com/story-apps/starc/releases/download/v${finalAttrs.version}/starc-setup.AppImage";
+    hash = "sha256-KAY04nXVyXnjKJxzh3Pvi50Vs0EPbLk0VgfZuz7MQR0=";
+  };
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase =
+    let
+      appimageContents = appimageTools.extract { inherit (finalAttrs) pname version src; };
+      starc-unwrapped = appimageTools.wrapType2 { inherit (finalAttrs) pname version src; };
+    in
+    ''
+      runHook preInstall
+
+      # Fixup desktop item icons
+      install -D ${appimageContents}/starc.desktop -t $out/share/applications/
+      sed -i -e 's:Icon=starc:Icon=dev.storyapps.starc:' \
```

Comment:
Could `substituteInPlace` be used here?

Reply:
Since I'm also using `sed`'s a command for appending, is this actually any better? I suppose I could substituteInPlace and then append in another step, if you like.